kong/api/routes/kong.lua

```diff
@@ -269,5 +269,20 @@ return {
       }
       return kong.response.exit(200, body)
     end
-  }
+  },
+  ["/status/dns"] = {
+    GET = function (self, db, helpers)
+      if not kong.dns.stats then
```

Comment:
TODO: thijs: kong.conf.legacy_dns_switch?

Reply:
hi @Tieske 

done https://github.com/Kong/kong/pull/12305/commits/36ae59a83a2e788ef39f6892f03f715902ab8d88, and added some test cases